app/assets/javascripts/school_administrator_dashboard/discipline_dashboard/SchoolDisciplineDashboard.js

```diff
@@ -186,22 +235,10 @@ export default class SchoolDisciplineDashboard extends React.Component {
   }
 
   renderStudentDisciplineTable(allIncidents, groupedIncidents) {
-    const students = this.groupStudents();
-    const studentDisciplineIncidentCounts = this.state.selectedCategory ? //if the user is looking at a subgroup of incidents
-                                            this.studentDisciplineIncidentCounts(groupedIncidents[this.state.selectedCategory]) :
-                                            this.studentDisciplineIncidentCounts(allIncidents);
-    let rows =[];
-    students.forEach((student) => {
-      rows.push({
-        id: student.id,
-        first_name: student.first_name,
-        last_name: student.last_name,
-        latest_note: student.latest_note,
-        events: studentDisciplineIncidentCounts[student.id] || 0,
-        grade: student.grade
-      });
-    });
-
+    const selectedIncidents = this.state.selectedCategory ? //if the user is looking at a subgroup of incidents
+                              groupedIncidents[this.state.selectedCategory] :
+                              allIncidents;
+    const rows = selectedIncidents ? this.studentDisciplineIncidentRows(selectedIncidents) : [];
```

Comment:
I'm wondering about how we can remove the work that `studentDisciplineIncidentRows` is doing in order to create a list of `[{student, disciplineIncidentCount}]` that respects the various filters.

Here's what I see here, let me know if I'm following.  Overall, this component gets a list of students with discipline incidents for each, and it stores in state the info about the various filters.  .  `render` flattens all the incidents and filters them and calls this method with the flattened incidents.  This method then has to call `studentDisciplineIncidentRows` to group those back together by students so it can count them.

Check out how this works in the [absences page](https://github.com/studentinsights/studentinsights/blob/master/app/assets/javascripts/school_absences/SchoolAbsencesDashboard.js#L150), where it's essentially:
```
this.filteredStudents().map(student => {
  return {
    ...student,
    absenceCount: this.filteredAbsences(student).length
  }
})
```
Would a similar approach help simplify here?

Reply:
Let me think about that. All the rows method is doing is taking a slice of the displayed incidents (based on the selected category) and reducing it to the information that the students table needs to display them. It does check to see if it should be using a subgroup, but it isn't changing or regrouping the flattened incidents. 

The main difference here between the absence dashboard is that we don't show students without discipline incidents so it makes more sense to think of this as a map of the discipline incidents with student data attached rather than a map of each student that we would then have to filter for students with no incidents.